docs/collection.js

```diff
@@ -158,9 +158,11 @@ class Collection {
      * });
      *
      * @param {string|Realm.Sync.SubscriptionOptions} options - an optional name or configuration object for the subscription.
-     *   The configuration object must contain `name` and the optional properties `update` and `timeToLive`.
+     *   The configuration object must contain `name` and the optional properties `update`, `timeToLive` and `inclusions`.
      *   If `update` is true the query on an existing subscription will be updated. If `timeToLive` is set, it indicates
-     *   to Realm for how long it must be kept alive, after which it will be automatically removed.
+     *   to Realm for how long it must be kept alive, after which it will be automatically removed. The `inclusions` property
+     *   is an array of property paths which specifies which linkingObjects relationships should be included in the subscription.
```

Comment:
It doesn't appear to be just linkingObjects properties that can be included in this array.

Reply:
`inclusions` is an incredibly vague name and seeing something like `.subscribe({inclusions: ['parents']})` wouldn't leave me with any clue about what it's doing.

Reply:
I'm struggling to come up with an intuitive name. How about `included_parents`, `included_reverse_relationships`, or `reverse_relationship_inclusions`? (cc: @kneth )

Reply:
Since we call the `linkingObjects` in the schema, maybe `includes_linkingObjects` is less confusing?

Reply:
In JS naming conventions are camelCase, but trying to weave in the word linkingObjects seems beneficial IMO. I don't have a strong opinion on using `include/includes/included`, but e.g. `includeLinkingObjects` seems somewhat descriptive.

Reply:
As I started on the Java API, what about `downloadLinkingObjects`?

Reply:
Personally, I think "download" is confusing terminology to add because it is not a comprehensive description of what is happening, we want to include these relationships in the subscription and keep them up to date as opposed to downloading them in the traditional "once off" sense of the word. Maybe I'm overthinking it. But I like `includeLinkingObjects`, because I think it reads like those relationships are included in the subscription (ie: `.subscribe({name: "myName", includeLinkingObjects: ['parents']})`)

Reply:
Side question, the recent updates of ttl and other parameters on the subscription parameters added a restriction that if the parameter to `subscribe` is an object, it must include a user defined name. This means that if users want to add a `includeLinkingObjects` parameter they  must also name the subscription. Is that an unnecessary restriction?

Reply:
`includeLinkingObjects` seems decent to me.

Reply:
I agree - all properties on the subscribe argument should be optional, including the name.

Reply:
The reason I made `name` required is that `update` doesn't make any sense if you don't have a name, as you don't know what to update. Idon't know If that weights stronger than allowing fetching linking objects on an unnamed subscription.

In any case, we can make a runtime check for it. It was just nicer when it was in the type system.

Reply:
Update without a name could probably work like Realm.create(update: true) for object without a primary key, or we could throw an error in that particular case 🤷‍♂️ I felt it's a bit annoying though as that meant you can't register an anonymous subscription that expires after some time.

Reply:
Sure, I don't have a problem lifting that restriction 👍

Reply:
Thanks for the input. I renamed to `includeLinkingObjects` and lifted the restriction requiring the name in the subscription options.